pkg/controller/cronjob/cronjob_controllerv2_test.go

```diff
@@ -1674,3 +1700,46 @@ func TestControllerV2GetJobsToBeReconciled(t *testing.T) {
 		})
 	}
 }
+
+func TestWatchCronJobs(t *testing.T) {
+	clientset := fake.NewSimpleClientset()
+	fakeWatch := watch.NewFake()
+	clientset.PrependWatchReactor("cronjobs", core.DefaultWatchReactor(fakeWatch, nil))
+	manager, sharedInformerFactory := newControllerFromClient(clientset, controller.NoResyncPeriodFunc)
+	manager.cronJobListerSynced = alwaysReady
+	manager.jobListerSynced = alwaysReady
+
+	var testCronJob batchv1.CronJob
+	received := make(chan struct{})
+
+	// The update sent through the fakeWatcher should make its way into the workqueue,
+	// and eventually into the syncHandler.
+	manager.syncHandler = func(ctx context.Context, cronJob *batchv1.CronJob, _ []*batchv1.Job) (*batchv1.CronJob, *time.Duration, bool, error) {
```

Comment:
I'm not sure I follow, why do you want to test watches? That's already handled by the tests in client-go, no?

Reply:
This should be covered by the e2e test for cronjob. This UT just confirmed that the `cronjob_controller` can be running and syncing.

I found that `func (jm *ControllerV2) Run(ctx context.Context, workers int)` is not covered in UT and just adding one simple `TestWatch` here.


I follow the style of `TestWatchJobs `. 
https://github.com/kubernetes/kubernetes/blob/4dd887797fe53816faf3cd4d65d600a606714f5d/pkg/controller/job/job_controller_test.go#L3618

Reply:
/priority backlog

Reply:
We don't have to cover every possible case in units, if you care about such a test, I'd suggest an integration, but I'm pretty confident we already cover this case. On top of that overwriting `syncHandler` isn't something I'm willing to accept.

Reply:
/close

per comments above.